src/Audio_details.md

```diff
@@ -0,0 +1,192 @@
+Thus, the counter can be made to increase faster by writing to `DIV` while its relevant bit is set (which clears `DIV`, and triggers the falling edge).
+
+Some events are tied to the DIV-APU counter:
+
+Event          | Every N ticks | Frequency[^div_apu_freq]
+---------------|---------------|-------------------------
+Envelope sweep | 8             | 64 Hz
+Sound length   | 2             | 256 Hz
+CH1 freq sweep | 4             | 128 Hz
+
+[^div_apu_freq]:
+Indicated values are under normal operation; the frequencies will obviously differ if writing to `DIV` to increase the counter faster.
+
+### Mixer
+
+A high-pass filter (HPF) removes constant biases over time.
+The HPFs therefore remove the DC offset created by inactive channels with an enabled DAC, and off-center waveforms.
+
+Enabling or disabling a DAC ([see below](#DACs)), adding or removing it using NR51, or changing the volume in NR50, will cause an audio pop.
+(All of these actions cause a change in DC offset, which is smoothed out by the HPFs over time, but still creates a pop.)
+
+The HPF is more aggressive on GBA than on GBC, which itself is more aggressive than on DMG.
+(The more "aggressive" a HPF, the faster it pulls the signal towards "analog 0"; this reduces the pops, but tends to also distort waveforms.)
+
+### DACs
+
+Channel <var>x</var>'s DAC is enabled if and only if `[NRx2] & $F8 != 0`; the exception is CH3, whose DAC is directly controlled by bit 7 of NR30 instead.
+Note that the envelope functionality changes the volume, but not the value stored in NRx2, and thus doesn't disable the DACs.
+
+If a DAC is enabled, the digital range $0 to $F is lineraly translated to the analog range -1 to 1, in arbitrary units.
+Importantly, the slope is negative: "digital 0" maps to "analog 1", not "analog -1".
+
+If a DAC is disabled, it fades to an analog value of 0.
+The nature of this fade is not entirely deterministic and varies between models.
+
+NR52's low 4 bits report whether the channels are turned on, not their DACs.
+
+### Channels
+
+A channel is activated by a write to NRx4's MSB, unless its DAC is off, which forces it to be disabled as well.
+The opposite is not true, however: a disabled channel outputs 0, which an enabled DAC will dutifully convert into "analog 1".
+
+A channel can be deactivated in one of the following ways:
+- Turning off its DAC
+- Its [length timer](<#Length timer>) expiring
+- (CH1 only) [Frequency sweep](<#FF10 — NR10: Channel 1 sweep>) overflowing the frequency
+
+### Pulse channels (CH1, CH2)
+
+Each pulse channel has an internal "duty step" counter, which is used to index into [the selected waveform](<#FF11 — NR11: Channel 1 length timer & duty cycle>) (each background stripe corresponds to one "duty step")[^pulse_lut].
```

Comment:
This doesn't read very well.. The internal counter is used to index (what?) into a waveform? What does it mean to index into a waveform?

Reply:
The "waveform" here is meant as an array of booleans, which is being indexed into.

Reply:
How is the counter influencing / "being indexed into" the waveform? Are you trying to say that it's being "encoded" in array used to generate the waveform?

Reply:
The waveform SVGs have stripes in the background, which are meant to indicate what the waveform's state is based on that "duty step"/phase counter. 0 being leftmost.